core/domain/fs_services_test.py

```diff
@@ -189,6 +189,37 @@ def test_save_original_and_compressed_versions_of_image(self) -> None:
         self.assertTrue(fs.isfile('image/%s' % self.COMPRESSED_IMAGE_FILENAME))
         self.assertTrue(fs.isfile('image/%s' % self.MICRO_IMAGE_FILENAME))
 
+    def test_skip_upload_if_image_already_exists(self) -> None:
+        with utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'), 'rb', encoding=None
+        ) as f:
+            original_image_content = f.read()
+        fs = fs_services.GcsFileSystem(
+            feconf.ENTITY_TYPE_EXPLORATION, self.EXPLORATION_ID)
+
+        # Save the image for the first time.
+        fs_services.save_original_and_compressed_versions_of_image(
+            self.FILENAME, 'exploration', self.EXPLORATION_ID,
+            original_image_content, 'image', True)
+
+        # Ensure the image is saved.
+        self.assertTrue(fs.isfile('image/%s' % self.FILENAME))
+
+        # Get the content of the saved image.
+        saved_image_content = fs.get('image/%s' % self.FILENAME)
+
+        # Save the image again (should be skipped due to existence).
+        fs_services.save_original_and_compressed_versions_of_image(
+            self.FILENAME, 'exploration', self.EXPLORATION_ID,
+            original_image_content, 'image', True)
+
+        # Get the content of the image after attempting the second save.
+        new_saved_image_content = fs.get('image/%s' % self.FILENAME)
+
+        # Check that the content of the image remains the same after the second
+        # save attempt.
+        self.assertEqual(saved_image_content, new_saved_image_content)
```

Comment:
@won369369 can we test if fs.commit is called?

Reply:
Hmm... I'm not sure. Can you give any suggestions? Thank you.

Reply:
Check this article out: https://thelearning.dev/pytest-how-to-test-that-a-function-is-called-with-specific-parameters

Reply:
Done!

Reply:
Sorry, if I was not clear. Can you test if `fs.commit` us called using `mock.patch` within the `test_skip_upload_if_image_already_exists` function?